src/sinks/new_relic/encoding.rs

```diff
@@ -22,7 +24,11 @@ impl Encoder<Result<NewRelicApiModel, NewRelicSinkError>> for NewRelicEncoder {
             writer.write_all(json)?;
             Ok(())
         })?;
-        io::Result::Ok(size)
+
+        // TODO This should not be zero.
+        let byte_size = telemetry().create_request_count_byte_size();
```

Comment:
What's the impact of this, and do we have any plan to resolve it?

Reply:
Yes. I've had to restructure the `new_relic` sink to move more functionality into the `Encoder` - I think it's more correct now and we can get the sizes correctly.

There are no integration tests for new relic so I haven't been able to test this fully, so a thorough check would be appreciated.